app/main.py

```diff
@@ -1,13 +1,30 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8",
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [
+            format_linter_error(err)
+            for err in errors
+            if err["filename"] == file_path
+        ],
+        "path": file_path,
+        "status":
+            "failed"
+            if [err for err in errors if err["filename"] == file_path]
```

Comment:
you just need to check if the list of errors is not empty, no need in the list comprehension